### PR TITLE
ath79: fix base configuration for Ubiquiti XM series

### PR DESCRIPTION
**Pull request by Leo-PL**
This series of patches brings the Ubiquiti XM series (Bullet-M, Nano-M, Rocket-M) to their default configuration based on information from ar71xx target.

The first patch allows EEPROM image loading for ath9k module from ART partition. This is crucial for wireless interface to operate. Result is correct MAC and correct SSID broadcast by onboard radio.
Second patch fixes default network configuration to create LAN interface only (only for Bullet-M and Rocket-M), as these two are single-port devices. Nano-M is kept as is.
Third patch adds back support for system status LED by updating device tree aliases.
Last patch makes the four RSSI LEDs present on board usable with "rssileds" package by creating default configuration.

The series was runtime and functionally tested on Ubiquiti Nanobridge M5 unit, using target bullet-m.

### CODE REVIEW

File: target/linux/ath79/base-files/etc/board.d/01_leds
Comment (mkresin) [inline target/linux/ath79/base-files/etc/board.d/01_leds]: I'm quite sure the last two parameters are not really used on your boards. These are used to change the brightness of pwm leds. According to your dts it looks like the rssi leds are "normal" ones.

File: target/linux/ath79/dts/ar7241_ubnt_xm.dtsi
Comment (mkresin) [inline target/linux/ath79/dts/ar7241_ubnt_xm.dtsi]: Since you're hijacking a led for a completely different purpose, you should limit the hijacking to the bare minimum. Please use the led only for boot and failsafe.

**Review by mkresin — CHANGES_REQUESTED** [review]
Please use ```git commit --amend``` to add changes to your commit. Use ```git push --force``` to overwrite your github repository/branch. The PR will afterwards update automatically. Please don't close this PR and open a new one.

File: target/linux/ath79/base-files/etc/board.d/01_leds
Comment (Leo-PL) [reply target/linux/ath79/base-files/etc/board.d/01_leds]: I'll fix and retest that later today. LEDs are indeed without PWM capability with regards to underlying hardware. Thanks for noticing.

File: target/linux/ath79/dts/ar7241_ubnt_xm.dtsi
Comment (Leo-PL) [reply target/linux/ath79/dts/ar7241_ubnt_xm.dtsi]: I'm a bit skeptical about this. This will likely change the external behaviour known from ar71xx target where this LED is used as status LED, since there is no dedicated externally visible status LED on those boards.

This mapping did not conflict with RSSI function when I added "rssileds" package. rssileds daemon did simply override the status LED, and IMO should not conflict with status LED even when entering upgrade.

I will test the changed mapping later today and post an update. Thanks you for noticing.

Comment (Leo-PL) [reply target/linux/ath79/dts/ar7241_ubnt_xm.dtsi]: I tested the requested change, but I found, that expectedly, after the boot phase, LED4 turns off instead of staying solid. This is different behaviour from ar71xx port, and I consider this a part of user interface, which I'd like to keep stable.

Of course, the final decision is up to you, so I'm waiting for feedback :)

Comment (mkresin) [reply target/linux/ath79/dts/ar7241_ubnt_xm.dtsi]: drop it. we had the discussion multiple times before. don't hijack leds longer than necessary. the relay important signalisations are:

- still booting
- time to hit a button to get into failsafe
- we are in failsafe

all of the written above can be done by keeping only the boot and the failsafe led.

In ar71xx a lot of hacks were introduced. we are working hard on not copying them to ath79.

Comment (Leo-PL) [reply target/linux/ath79/dts/ar7241_ubnt_xm.dtsi]: OK, Understood. I'll post an update.

**Comment by mkresin** [comment]
merged, thanks!

**Pull request closed without merging**
